File: archive/c/c/even-odd.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{    
    if (argc == 1 || sizeof(argv[1]) == 0 || (atoi(argv[1]) == 0 && !strcmp(argv[1], "0"))) {
        printf("Usage: please input a number\n");
    } else {
        int input = atoi(argv[1]);
        if (input % 2 == 0) {
            printf("Even\n");
        }
        else {
            printf("Odd\n");
        }
    }

    return 0;
}

File: archive/r/r/even-odd.R
args<-commandArgs(TRUE)
number = args[1]

if (length(args) < 1 || !suppressWarnings(!is.na(as.numeric(number)))) {
  cat("Usage: please input a number")
}else {
  number = as.numeric(number)
  if (number %% 2 == 0) {
    cat("Even")
  }else {
    cat("Odd")
  }
}
